### PR TITLE
Frontend Registration Recommendations

### PR DESCRIPTION
**Pull request by cemreefe**
* I changed the input fields for `university` and `department` into single selects.
* I changed the name of the `degree` field into `title` in `infoUpdate` requests, in the form and also in the collaborators sider in project-details page.
* I added necessary API call functions to get recommendations from the getTags, getUniversities and getDepartments endpoints of the backend instance. It is called `recommendations` and its related functions/entries can be found in the files in `redux/search`.

### CODE REVIEW

File: frontend/src/pages/SignUp/index.jsx [hunk 1]
```diff
@@ -78,6 +53,46 @@ const SignUp = () => {
...
+    } else if(formStep===1){
+      console.log('1');
+    } else if(formStep===2){
+      console.log('2');
+      dispatch(recommendations('tag', setTags));
+      dispatch(recommendations('uni', setUniversities));
+      dispatch(recommendations('dep', setDepartments));
+    }
+
+  },[formStep]);
```
Comment (aliramazanmert) [inline frontend/src/pages/SignUp/index.jsx:69]: need to delete console.logs and dispatch seems not to be using actual redux reducer, instead sending the setters. We should probably refactor that

File: frontend/src/pages/SignUp/index.jsx [hunk 2]
```diff
@@ -78,6 +53,46 @@ const SignUp = () => {
...
+    for (let i = 0; i < universities.length; i++) {
+      uc.push(<Option key={universities[i]}>{universities[i]}</Option>);
+    }
+    for (let i = 0; i < departments.length; i++) {
+      dc.push(<Option key={departments[i]}>{departments[i]}</Option>);
+    }
+    setInterestChoices(ic);
+    setUniversityChoices(uc);
+    setDepartmentChoices(dc);
+  },[tags, universities, departments]);
```
Comment (aliramazanmert) [inline frontend/src/pages/SignUp/index.jsx:92]: Didn't really understand your purpose here

File: frontend/src/redux/search/api.js [hunk 3]
```diff
@@ -12,4 +12,36 @@ export const search = (params, setData, setLoading) => {
...
+    dispatch(actions.recommendationsAction());
+    console.log('x');
+
+    api({sendToken: true})
+    .get(dispatchUrl)
+    .then((response) => {
+      setRecommendations(response.data.result);
+      console.log('c');
+      console.log(response.data.result);
+      dispatch(actions.recommendationsCompletedAction());
```
Comment (aliramazanmert) [inline frontend/src/redux/search/api.js:44]: console logs need to go. Again, we should set the redux state not the component's state. This logic should warn you in the browser console as well, with something like "you can't change the state of an unmounted component".

File: frontend/src/routes.jsx [hunk 4]
```diff
@@ -35,6 +36,9 @@ export default function App() {
         setLoading(false);
       });
     // eslint-disable-next-line
+
+    dispatch(recommendationsAction());
+
```
Comment (aliramazanmert) [inline frontend/src/routes.jsx:41]: why is this here ?

**Review by aliramazanmert — CHANGES_REQUESTED** [review]
Some minor changes needed in different cases and some changes needed in the redux reducer state logic.

File: frontend/src/pages/SignUp/index.jsx (hunk 2 above)
Comment (cemreefe) [reply frontend/src/pages/SignUp/index.jsx:92]: the multiselect antd component takes a list of `<Option />` objects, while, the endpoints return mere lists. I take those lists from the api returns and create `<Option />` object lists.

File: frontend/src/routes.jsx (hunk 4 above)
Comment (cemreefe) [reply frontend/src/routes.jsx:41]: no idea, deleting

File: frontend/src/pages/SignUp/index.jsx (hunk 1 above)
Comment (cemreefe) [reply frontend/src/pages/SignUp/index.jsx]: Done.

File: frontend/src/pages/SignUp/index.jsx (hunk 2 above)
Comment (cemreefe) [reply frontend/src/pages/SignUp/index.jsx]: We have edited this code to avoid unnecessary overhead on api call results. Resolving conversation.

**Review by aliramazanmert — APPROVED** [review]
we can merge now !

**Comment by cemreefe** [comment]
Now the pre-defined values for `Tags`, `Departments` and `Universities` fields are taken from the backend.
